lib/epochs.js

```diff
@@ -105,6 +105,55 @@ function Epochs(ssb) {
     })
   }
 
+  function getTipEpochs(groupId, cb) {
+    if (cb === undefined) return p(this.getTipEpochs).call(this, groupId)
+
+    const opts = { getters: pluck(allGetters, ['author', 'secret']) }
+    epochsReduce(groupId, opts, (err, reduce) => {
+      // prettier-ignore
+      if (err) return cb(clarify(err, 'Failed to resolve epoch @tangle/reduce'))
+
+      const tips = Object.keys(reduce.state).map((id) => {
+        const info = {
+          id,
+          previous: reduce.graph.getNode(id).previous,
+          ...reduce.state[id][id],
+        }
+        info.secret = Buffer.from(info.secret, 'base64')
+        return info
+      })
+
+      return cb(null, tips)
+    })
+  }
+
+  /**  Gets all epochs leading up to the specified epoch, all the way back to the root.
+   * Does not include the specified epoch nor epochs in epoch forks.
+   */
+  function getPredecessorEpochs(groupId, epochRootId, cb) {
+    if (cb === undefined)
+      return p(this.getPredecessorEpochs).call(this, groupId, epochRootId)
+
+    const opts = { getters: pluck(allGetters, ['author', 'secret']) }
```

Comment:
you're getting the author data but not using it :shrug: 
(doesn't really cost much)